daemon/main.go

```diff
@@ -108,6 +109,12 @@ var RootCmd = &cobra.Command{
 }
 
 func main() {
+
+	// Open socket for using gops to get stacktraces of the agent.
+	if err := gops.Listen(gops.Options{}); err != nil {
```

Comment:
Add this only in debug.

Reply:
I disagree. This is a production troubleshooting tool, not a debug tool. This must be always on, in both development and production. It has zero overhead, so I don't see any reason not to do this.

Reply:
I'm with Romain on this - we need this precisely when we hit an issue and need diagnostics.